lua/pantran/curl.lua

```diff
@@ -22,9 +22,14 @@ function curl:_spawn(request, path, data, callback)
     tostring(self._url / path)
   })
 
-  for key, value in pairs(vim.tbl_extend("error", self._data, data)) do
-    table.insert(args, 1, ("%s=%s"):format(key, value))
-    table.insert(args, 1, "--data-urlencode")
+  if self._headers["Content-Type"] == "application/json" then
+    table.insert(args, 1, vim.json.encode(vim.tbl_extend("error", self._data, data)))
```

Comment:
How about using `vim.tbl_extend("error", vim.empty_dict(), self._data, data)` here, instead of doing it in line 126?

Reply:
You are right. Done.